src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(value) {
+    if (value !== undefined) {
+      sum += value;
+
+      return adder;
+    } else {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
```

Comment:
```suggestion
    if (value !== undefined) {
      sum += value;

      return adder;
    }
    
    const result = sum;

    sum = 0;

    return result;
```
block else is redundant here